std/haxe/macro/Expr.hx

```diff
@@ -573,6 +573,26 @@ enum ComplexType {
 }
 
 /**
+	Represents an argument of a function type (`ComplexType.TFunction`).
+**/
+typedef FunctionTypeArg = {
```

Comment:
If we add a new type anyway we should consider adding the name position.

Reply:
I wanted to ask about that, don't we want to have generic `PlacedName` instead of adding `namePos` fields?

Reply:
I'm not sure. I think a flat structure is easier to handle, less breaking and even more performant (no nested object allocation).

Reply:
Should I also add `typePos` then?

TBH I wouldn't do that just now because adding positions to random types sounds really inconsistent. I would rather wait for some kind of parse tree integration to support precise positions and leave AST alone.

Reply:
My concern is that this will require people to update their macro code, and then if/when we add more positions, they have to update it again.

Also, we definitely have to make sure we don't lose the position in an encode/decode roundtrip. Check the other occurrences of `encode_placed_name`, they all come with a `name_pos` encoding as well. It's just not exposed on the Haxe side yet.

Reply:
Alright, I added `namePos` (should be that instead of `name_pos` for Haxe, right?)